File: app/src/main/java/com/chiului/android_mvvm_architecture/data/AppCacheDao.java
package com.chiului.android_mvvm_architecture.data;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.chiului.android_mvvm_architecture.bean.AppCacheBean;

/**
 * $AppCache的DAO(数据访问对象)
 *
 * @author
 * @date 2020/07/23 10:50
 */
@Dao
public interface AppCacheDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAppCache(AppCacheBean cache);

    @Delete
    void deleteAppCache(AppCacheBean cache);

    @Query("SELECT * FROM app_cache WHERE appKey == :key")
    AppCacheBean getAppCache(String key);

}

File: app/src/main/java/com/chiului/android_mvvm_architecture/data/AppCacheDao.kt
package com.chiului.android_mvvm_architecture.data

import androidx.room.*
import com.chiului.android_mvvm_architecture.bean.AppCacheBean

/**
 * AppCache 的 DAO(数据访问对象)
 * @author    神经大条蕾弟
 * @date      2021/02/01 11:09
 */
@Dao
interface AppCacheDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAppCache(cache: AppCacheBean)

    @Delete
    fun deleteAppCache(cache: AppCacheBean)

    @Query("SELECT * FROM app_cache WHERE `key` == :key")
    fun getAppCache(key: String): AppCacheBean?

}